include/support.h

```diff
@@ -326,6 +326,12 @@ bool contains(const container_t &container, const typename container_t::key_type
 	return container.find(key) != container.end();
 }
 
+template <typename container_t>
+bool set_contains(const container_t &container, const typename container_t::key_type &key)
+{
+	return container.find(key) != container.end();
+}
+
```

Comment:
If anyone can figure out how to make `std::set` work with the `contains()` overload, this won't be needed.

Problem is that sets have the same key and value types.

Reply:
`set` is strictly the right data structure (guarantees unique elements, be it one element or millions of elements).

But in this case, we're dealing with a handful of elements, so the [points made here](http://lafstern.org/matt/col1.pdf) saying, _"why we shouldn't use <set> for small quantities"_ apply.  (Yeah - this is sounding pedantic! but I'm trying to find a way that we can avoid set all together here).

This code already uses a `vector<string>` instance, so that's a "sunk cost" that we can basically get for free, instead of adding all of `<set>` to the pile:

```c++
	std::vector<std::string> startup_params = {};
	std::vector<std::string> configfiles = {};
	std::set<std::string> loadedConfigFIles = {};
```

All of that say, suggest we just use a vector.